packages/backend-common/src/service/lib/ServiceBuilderImpl.ts

```diff
@@ -40,7 +40,7 @@ export class ServiceBuilderImpl implements ServiceBuilder {
   private host: string | undefined;
   private logger: Logger | undefined;
   private corsOptions: cors.CorsOptions | undefined;
-  private routers: [string, Router][];
+  private routers: [string, Router | RequestHandler][];
```

Comment:
Not opposed to this addition, but thinking it might actually be easier to implement it with the promise router + a single `.use((req, res) => ...)`

Reply:
My first iteration actually wrapped this in a `router` and treated is similar to how plugin routers are registered. This would work, but would require nested a sub route `/[plugin-root]/healthcheck`. The middleware as is very flexible and could be used for plugin specific checks, etc. That said, there could also be a `router` that wraps this and is re-usable as well. However, currently the `ServiceBuilderImpl` makes assumption as to how things are registered and does not allow for much deviation unless you wanted to do it all from scratch. Might need to think about in what ways it should be customizable.

Reply:
@Rugvip - Can you comment on this ^^. I'm very open to alternatives here. This could set a precedent / pattern, so I think establishing the baseline is important in regard to how the `statusCheckHandler` is and should be integrated / configured.

Reply:
Thinking it would be the equivalent of `.use(await statusCheckHandler())`, so probably `.addRouter('/', await statusCheckHandler())`? Not sure if the `'/'` works though, might have to be `''`. Another option would be to add a single-arg overload of `.addRouter`, and maybe even just rename it to `.use` while at it :grin:

Reply:
@Rugvip - I opted to create a `createStatusCheckRouter` function in `backend-common` which utilizes the `statusCheckHandler` and revert the changes to `ServiceBuilderImpl` for now. I didn't want to create a breaking change as part of this PR. However, the building blocks are here to be able to use either approach which some tweaks to the `ServiceBuilder` now.